src/main/java/org/prebid/server/bidder/bidstack/BidstackBidder.java

```diff
@@ -0,0 +1,159 @@
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    private static final TypeReference<ExtPrebid<?, ExtImpBidstack>> BIDDER_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    public BidstackBidder(String endpointUrl,
+                          CurrencyConversionService currencyConversionService,
+                          JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.currencyConversionService = Objects.requireNonNull(currencyConversionService);
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> updatedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final BigDecimal resolvedBidFloor = resolveBidFloor(request, imp);
+                updatedImps.add(updateImp(imp, resolvedBidFloor));
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest updatedBidRequest = updateBidRequest(request, updatedImps);
+
+        return Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(constructHeaders(updatedBidRequest))
+                .body(mapper.encodeToBytes(updatedBidRequest))
+                .payload(updatedBidRequest)
+                .build());
+    }
+
+    private ExtImpBidstack parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper()
+                    .convertValue(imp.getExt(), BIDDER_EXT_TYPE_REFERENCE)
+                    .getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Unable to decode the impression ext for id: " + imp.getId());
+        }
+    }
+
+    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
+        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
+                ? convertBidFloorCurrency(imp.getBidfloor(), request, imp.getId(), imp.getBidfloorcur())
+                : imp.getBidfloor();
+    }
+
+    private Imp updateImp(Imp imp, BigDecimal bidFloor) {
+        return imp.toBuilder()
+                .bidfloorcur(BIDDER_CURRENCY)
+                .bidfloor(bidFloor)
+                .ext(mapper.mapper().createObjectNode()
+                        .set("bidder", mapper.mapper().valueToTree(parseExtImp(imp))))
+                .build();
+    }
+
+    private static BidRequest updateBidRequest(BidRequest request, List<Imp> updatedImps) {
+        return request.toBuilder()
+                .imp(updatedImps)
+                .cur(Collections.singletonList(BIDDER_CURRENCY))
```

Comment:
Same. I don't see any similar logic in GO.

Reply:
Do you mean that we don't need this part?
```
  "cur": [
    "USD"
  ],
```